tests/run-make/mixing-formats/rmake.rs

```diff
@@ -0,0 +1,93 @@
+// Testing various mixings of rlibs and dylibs. Makes sure that it's possible to
+// link an rlib to a dylib. The dependency tree among the file looks like:
+//
+//                 foo
+//               /     \
+//             bar1   bar2
+//             /    \ /
+//          baz    baz2
+//
+// This is generally testing the permutations of the foo/bar1/bar2 layer against
+// the baz/baz2 layer
+
+//@ ignore-cross-compile
+
+use run_make_support::{rustc, tmp_dir};
+use std::fs;
+
+fn test_with_teardown(rustc_calls: &dyn Fn()) {
```

Comment:
Nit: can't this just be `f: Fn()`, it doesn't need to be `&dyn Fn()` AFAICT?

Reply:
I tried it that way initially, but this triggers the following error message:

```
2024-05-22T18:08:36.1683386Z --- stderr -------------------------------
2024-05-22T18:08:36.1684109Z error[E0277]: the size for values of type `(dyn Fn() + 'static)` cannot be known at compilation time
2024-05-22T18:08:36.1703969Z ##[error]  --> /checkout/tests/run-make/mixing-formats/rmake.rs:18:23
2024-05-22T18:08:36.1711694Z    |
2024-05-22T18:08:36.1712020Z 18 | fn test_with_teardown(rustc_calls: Fn()) {
2024-05-22T18:08:36.1712621Z    |                       ^^^^^^^^^^^ doesn't have a size known at compile-time
2024-05-22T18:08:36.1713013Z    |
2024-05-22T18:08:36.1713456Z    = help: the trait `Sized` is not implemented for `(dyn Fn() + 'static)`
2024-05-22T18:08:36.1714031Z    = help: unsized fn params are gated as an unstable feature
2024-05-22T18:08:36.1714497Z help: you can use `impl Trait` as the argument type
2024-05-22T18:08:36.1714858Z    |
2024-05-22T18:08:36.1715140Z 18 | fn test_with_teardown(rustc_calls: impl Fn()) {
2024-05-22T18:08:36.1715523Z    |                                    ++++
2024-05-22T18:08:36.1716122Z help: function arguments must have a statically known size, borrowed types always have a known size
2024-05-22T18:08:36.1716913Z    |
2024-05-22T18:08:36.1717197Z 18 | fn test_with_teardown(rustc_calls: &dyn Fn()) {
2024-05-22T18:08:36.1717580Z    |                                    ++++
2024-05-22T18:08:36.1717817Z 
```
[Full log](https://productionresultssa12.blob.core.windows.net/actions-results/12027829-8ac3-45ae-a704-1c7b51b62256/workflow-job-run-7ff3f0f4-11f0-51c6-1fac-a5bd0881dbeb/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-05-22T22%3A27%3A16Z&sig=926eFVug3GU5tmhqr2kEwlEPABe3zzLmdJ48z8TA0n4%3D&sp=r&spr=https&sr=b&st=2024-05-22T22%3A17%3A11Z&sv=2021-12-02)

Reply:
I'm dumb and forgor the `impl` lol (sorry for misleading)

```rs
fn foo(f: impl Fn()) {
    f()
}

fn main() {
    foo(|| { println!("hello world") })
}
```